skyportal/handlers/api/candidate.py

```diff
@@ -1124,7 +1189,25 @@ def grab_query_results(
 
     items = []
     query_options = [joinedload(Obj.thumbnails)] if include_thumbnails else []
-    for item_id in page_ids:
-        items.append(Obj.query.options(query_options).get(item_id))
+
+    if len(obj_ids_in_page) > 0:
+        obj_ids_values = get_obj_id_values(obj_ids_in_page)
+        items = (
+            DBSession()
+            .query(Obj)
+            .options(query_options)
+            .join(obj_ids_values, obj_ids_values.c.id == Obj.id)
+            .order_by(obj_ids_values.c.ordering)
+        )
+    else:
+        # If there are no values, the VALUES statement above will cause a syntax error,
+        # so just switch to this simple IN query. Should return nothing either way
+        items = (
+            DBSession()
+            .query(Obj)
+            .options(query_options)
+            .filter(Obj.id.in_(obj_ids_in_page))
+        )
```

Comment:
Then why not just set `items = []` or `None` (whatever is expected downstream)?

Reply:
Downstream expects a Query object to be able to make `.add_columns()` calls on. This was the way I chose to fix because it was the quickest, but it could easily just be a `None` here and then do a bunch of `if not None` checks or something like that further downstream. That would probably be the more "correct" way of doing it tbh

Reply:
Ok I just pushed a commit to simplify this (just returns `[]` for items if no results) and handle it downstream